src/agent/src/cdh.rs

```diff
@@ -63,6 +83,84 @@ impl CDHClient {
         }
         Ok((*env.to_owned()).to_string())
     }
+
+    pub async fn unseal_file(&self, sealed_source_path: &String) -> Result<()> {
+        if !Path::new(sealed_source_path).exists() {
+            info!(
+                sl(),
+                "sealed source path {:?} dose not exists", sealed_source_path
+            );
+            return Ok(());
+        }
+
+        for entry in fs::read_dir(sealed_source_path)? {
+            let entry = entry?;
+
+            if !entry.file_type()?.is_symlink()
+                && !fs::metadata(entry.path())?.file_type().is_file()
+            {
+                info!(
+                    sl(),
+                    "skipping sealed source entry {:?} because its file type is {:?}",
+                    entry,
+                    entry.file_type()?
+                );
+                continue;
+            }
+
+            let target_path = fs::canonicalize(&entry.path())?;
+            info!(sl(), "sealed source entry target path: {:?}", target_path);
+            if !target_path.is_file() {
+                info!(sl(), "sealed source is not a file: {:?}", target_path);
+                continue;
+            }
+
+            let secret_name = entry.file_name();
+            let mut file = fs::File::open(&target_path)?;
+            let mut contents = String::new();
+            file.read_to_string(&mut contents)?;
```

Comment:
nit: could we replace these lines by using the convenience function, so something like:
```
let contents = fs::read_to_string(target_path)?;
```
?

Reply:
fixed, thanks